cluster-autoscaler/utils/kubernetes/listers.go

```diff
@@ -172,27 +172,31 @@ func UnschedulablePods(allPods []*apiv1.Pod) []*apiv1.Pod {
 	return unschedulablePods
 }
 
-// AllPodLister lists all pods.
-type AllPodLister struct {
+// ScheduledPodLister lists all scheduled pods.
+type ScheduledPodLister struct {
 	podLister v1lister.PodLister
 }
 
 // List returns all scheduled pods.
-func (lister *AllPodLister) List() ([]*apiv1.Pod, error) {
-	return lister.podLister.List(labels.Everything())
-}
+func (lister *ScheduledPodLister) List() ([]*apiv1.Pod, error) {
+	var pods []*apiv1.Pod
 
-// NewAllPodLister builds AllPodLister
-func NewAllPodLister(kubeClient client.Interface, stopchannel <-chan struct{}) PodLister {
-	selector := fields.ParseSelectorOrDie("status.phase!=" +
-		string(apiv1.PodSucceeded) + ",status.phase!=" + string(apiv1.PodFailed))
-	podListWatch := cache.NewListWatchFromClient(kubeClient.CoreV1().RESTClient(), "pods", apiv1.NamespaceAll, selector)
-	store, reflector := cache.NewNamespaceKeyedIndexerAndReflector(podListWatch, &apiv1.Pod{}, time.Hour)
-	podLister := v1lister.NewPodLister(store)
-	go reflector.Run(stopchannel)
+	allPods, err := lister.podLister.List(labels.Everything())
+	if err != nil {
+		return pods, err
+	}
+	for _, p := range allPods {
+		if p.Status.Phase != apiv1.PodSucceeded && p.Status.Phase != apiv1.PodFailed {
```

Comment:
Yes, because pod informer is shared in here and https://github.com/kubernetes/autoscaler/pull/6110/files#diff-5818325464670a5cd54280fe253b0841a163da5de2764491a3065805785ffb1bR47. We don't know if other places would need failed/succeeded pods or not. However, I believe failed/succeeded pods in practice are less due to pod GC so it should not be an issue for memory overhead

Reply:
Ok, makes sense.